### PR TITLE
Andrei/liniting

### PR DESCRIPTION
**Pull request by AndreiBozantan**
Enabled code formatting using Omnisharp tools.
Enabled code linting using Roslyn analyzers; Omnisharp tools will also continuously display the analyzers' results in VSCode.

### CODE REVIEW

File: src/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+# top-most EditorConfig file
+root = true
+
+# Don't use tabs for indentation.
+[*]
+indent_style = space
+# (Please don't specify an indent_size here; that has too many unintended consequences.)
+
+# Code files
+[*.{cs,csx,vb,vbx}]
```
Comment (sebastian) [inline src/.editorconfig]: do we need `vb` and `vbx`?

File: src/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+[*.{props,targets,ruleset,config,nuspec,resx,vsixmanifest,vsct}]
+indent_size = 2
+
+# JSON files
+[*.json]
+indent_size = 2
+
+# Powershell files
+[*.ps1]
+indent_size = 2
```
Comment (sebastian) [inline src/.editorconfig]: I doubt we'll have any powershell

File: src/.editorconfig (hunk 1 above)
Comment (AndreiBozantan) [reply src/.editorconfig]: I copied the [.editorconfig](https://github.com/dotnet/roslyn/blob/master/.editorconfig) file used by Roslyn team, as recommended [here](https://www.hanselman.com/blog/EditorConfigCodeFormattingFromTheCommandLineWithNETCoresDotnetFormatGlobalTool.aspx). I think that we can modify this as we see fit for our project.

Comment (AndreiBozantan) [reply src/.editorconfig]: Regarding the .editorconfig plugin for VSCode, I am not sure if it is necessary or not. From my understanding, the Omnisharp tools are handling the integration between VSCode and the editorconfig file with the help of the [ms-vscode.csharp](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp) extension. From my local tests, the integration seems to be working fine, at least for the simple things, like the indent size, the editor will reformat the file according to the editorconfig settings, when saving the file.

**Comment by AndreiBozantan** [comment]
@sebastian I added some more changes to this pull request, please check it again

File: src/.editorconfig [hunk 3]
```diff
@@ -107,7 +107,7 @@ dotnet_naming_rule.instance_fields_should_be_camel_case.style = instance_field_s
 dotnet_naming_symbols.instance_fields.applicable_kinds = field
 
 dotnet_naming_style.instance_field_style.capitalization = camel_case
-dotnet_naming_style.instance_field_style.required_prefix = _
+# dotnet_naming_style.instance_field_style.required_prefix = _
```
Comment (sebastian) [inline src/.editorconfig]: please remove it altogether instead

Comment (sebastian) [reply src/.editorconfig]: can be removed in a subsequent pull

Comment (sebastian) [reply src/.editorconfig]: i.e. does not need changes to this pull atm

**Pull request merged**